Expensify/App
# [No QA] Improve logging for Pusher `error` event. Avoid unnecessary reauthentication. Update Pusher.

### Details

- Improve logging for Pusher
- Stop triggering re-authentication when we handle a Pusher `error` event
- Pusher `error` events are not related to authentication and trigger when there are issues with websockets

### Fixed Issues
$ https://github.com/Expensify/App/issues/8683

### Tests
1. Sign into NewDot and grep JS logs for 'pusher'
2. Manually break your authToken somehow e.g. 
```diff
diff --git a/src/libs/Network/NetworkStore.js b/src/libs/Network/NetworkStore.js
index c0973fa50..c1cb1ff96 100644
--- a/src/libs/Network/NetworkStore.js
+++ b/src/libs/Network/NetworkStore.js
@@ -11,6 +11,8 @@ let hasReadRequiredData = false;
 let isAuthenticating = false;
 let isOffline = false;

+window.borkToken = () => authToken = 'piza';
+
 /**
  * @param {Boolean} val
  */
```
3. Go offline by shutting wifi off and wait about 15-30 seconds
4. Turn wifi back on
5. Verify that the logs show and that we are successfully connected to Pusher and the channels
![2022-04-18_12-57-24](https://user-images.githubusercontent.com/32969087/163890369-4cdfee15-7996-4412-acdf-bce310ca2d0f.png)


<details>
<summary><h4>PR Reviewer Checklist</h4></summary>


</details>

### QA Steps
No QA


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
